stan/math/rev/fun/quad_form.hpp

```diff
@@ -93,6 +94,65 @@ class quad_form_vari : public vari {
 
   quad_form_vari_alloc<Ta, Ra, Ca, Tb, Rb, Cb>* impl_;
 };
+
+/**
+ * Return the quadratic form \f$ B^T A B \f$.
+ *
+ * Symmetry of the resulting matrix is not guaranteed due to numerical
+ * precision.
+ *
+ * @tparam Mat1 type of the first (square) matrix
+ * @tparam Mat2 type of the second matrix
+ *
+ * @param A square matrix
+ * @param B second matrix
+ * @param symmetric indicates whether the output should be made symmetric
+ * @return The quadratic form
+ * @throws std::invalid_argument if A is not square, or if A cannot be
+ * multiplied by B
+ */
+template <typename Mat1, typename Mat2,
+          require_all_matrix_t<Mat1, Mat2>* = nullptr,
+          require_any_var_matrix_t<Mat1, Mat2>* = nullptr>
+inline auto quad_form_impl(const Mat1& A, const Mat2& B, bool symmetric) {
+  check_square("quad_form", "A", A);
+  check_multiplicable("quad_form", "A", A, "B", B);
+
+  using return_t
+      = promote_var_matrix_t<decltype(value_of(B).transpose().eval()
+                                      * value_of(A) * value_of(B).eval()),
+                             Mat1, Mat2>;
```

Comment:
Yeah, I was a little nervous about compile time sized matrices, which is why I kept the template.

In that case I guess I could have just done `var_value<decltype(value_of(B).transpose().eval() * value_of(A) * value_of(B).eval())>`.

Reply:
> In that case I guess I could have just done `var_value<decltype(value_of(B).transpose().eval() * value_of(A) * value_of(B).eval())>`.

Yes though you want

`var_value<plain_type_t<decltype(value_of(B).transpose().eval() * value_of(A) * value_of(B).eval())>>` because `var_value<>` can accept expressions as the template (For slices such as `Block<>`

Reply:
> `var_value<>` can accept expressions

Oh right. Can't let the guard down for a second on this stuff :/